cores/arduino/Print.cpp

```diff
@@ -286,10 +286,23 @@ size_t Print::printLongLong(unsigned long long n, uint8_t base) {
 }
 
 
-size_t Print::printFloat(double number, uint8_t digits)
+size_t Print::printFloat(double number, uint16_t decimalPlaces)
 {
-  char str[50];
-
-  dtostrf(number, 0, digits, str);
+    if(decimalPlaces>16)
+        decimalPlaces = 16;
```

Comment:
Magic number; can you stick this in a #define somewhere? i.e. MAX_DTOSTRF_DIGITS or something